sonar-plugin-api/src/main/java/org/sonar/api/ce/posttask/PostProjectAnalysisTaskTester.java

```diff
@@ -0,0 +1,530 @@
+  private static final String PROJECT_CAN_NOT_BE_NULL = "project can not be null";
+  private static final String CE_TASK_CAN_NOT_BE_NULL = "ceTask can not be null";
+
+  private final PostProjectAnalysisTask underTest;
+  @CheckForNull
+  private CeTask ceTask;
+  @CheckForNull
+  private Project project;
+  @CheckForNull
+  private Date date;
+  @CheckForNull
+  private QualityGate qualityGate;
+
+  private PostProjectAnalysisTaskTester(PostProjectAnalysisTask underTest) {
+    this.underTest = requireNonNull(underTest, "PostProjectAnalysisTask instance can not be null");
+  }
+
+  public static PostProjectAnalysisTaskTester of(PostProjectAnalysisTask underTest) {
+    return new PostProjectAnalysisTaskTester(underTest);
+  }
+
+  public static CeTaskBuilder newCeTaskBuilder() {
+    return new CeTaskBuilder();
+  }
+
+  public static ProjectBuilder newProjectBuilder() {
+    return new ProjectBuilder();
+  }
+
+  public static QualityGateBuilder newQualityGateBuilder() {
+    return new QualityGateBuilder();
+  }
+
+  public static ConditionBuilder newConditionBuilder() {
+    return new ConditionBuilder();
+  }
+
+  public PostProjectAnalysisTaskTester withCeTask(CeTask ceTask) {
+    this.ceTask = requireNonNull(ceTask, CE_TASK_CAN_NOT_BE_NULL);
+    return this;
+  }
+
+  public PostProjectAnalysisTaskTester withProject(Project project) {
+    this.project = requireNonNull(project, PROJECT_CAN_NOT_BE_NULL);
+    return this;
+  }
+
+  public PostProjectAnalysisTaskTester at(Date date) {
+    this.date = requireNonNull(date, DATE_CAN_NOT_BE_NULL);
+    return this;
+  }
+
+  public PostProjectAnalysisTaskTester withQualityGate(@Nullable QualityGate qualityGate) {
+    this.qualityGate = qualityGate;
+    return this;
+  }
+
+  public void execute() {
+    this.ceTask = requireNonNull(ceTask, CE_TASK_CAN_NOT_BE_NULL);
+    this.project = requireNonNull(project, PROJECT_CAN_NOT_BE_NULL);
+    this.date = requireNonNull(date, DATE_CAN_NOT_BE_NULL);
+
+    this.underTest.finished(
+      new PostProjectAnalysisTask.ProjectAnalysis() {
+        @Override
+        public CeTask getCeTask() {
+          return ceTask;
+        }
+
+        @Override
+        public Project getProject() {
+          return project;
+        }
+
+        @Override
+        public QualityGate getQualityGate() {
+          return qualityGate;
+        }
+
+        @Override
+        public Date getDate() {
+          return date;
+        }
+
+        @Override
+        public String toString() {
+          return "ProjectAnalysis{" +
+            "ceTask=" + ceTask +
+            ", project=" + project +
+            ", date=" + date.getTime() +
+            ", qualityGate=" + qualityGate +
+            '}';
+        }
+      });
+
+  }
+
+  @Beta
+  public static final class CeTaskBuilder {
+    private static final String ID_CAN_NOT_BE_NULL = "id can not be null";
+    private static final String STATUS_CAN_NOT_BE_NULL = "status can not be null";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "status can not be null" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)